### PR TITLE
Update custom reddit provider to v4

### PR DESCRIPTION
**Pull request by dgrcode**
## ☕️ Reasoning
The docs around Reddit provider were obsolete because they were using a previous version of NextAuth. That resulted in a weird error that led me to the discussion https://github.com/nextauthjs/next-auth/discussions/3204. It seems there was a bug open in a previous repo to update the docs, but the bug seems to have been lost on a repo migration.

In the scopes I've added the same scopes that were previously in the docs, but I've personally only tested with the "identity" scope. With this I mean that the code I've written might not work out of the box. However, given the previous code didn't work either, at worst we're still improving the docs.

## 🧢 Checklist

- [x] Documentation
- [ ] Tests
- [x] Ready to be merged

## 🎫 Affected issues

I haven't found any open issue searching for "reddit"

## 📌 Resources

- [Contributing guidelines](./CONTRIBUTING.md)
- [Code of conduct](./CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

File: docs/docs/providers/reddit.md [hunk 1]
```diff
@@ -47,25 +47,28 @@ This Provider template only has a one hour access token to it and only has the "
...
-      "https://www.reddit.com/api/v1/authorize?response_type=code&duration=permanent",
-    profileUrl: "https://oauth.reddit.com/api/v1/me",
-    profile: (profile) => {
-      return {
-        id: profile.id,
-        name: profile.name,
-        email: null,
-      }
-    },
+    id: 'reddit',
```
Comment (balazsorban44) [inline docs/docs/providers/reddit.md:50]: Let's use `RedditProvider`, so you only need to override what's different from the default config.

Comment (dgrcode) [reply docs/docs/providers/reddit.md]: sounds good 👍

**Comment by dgrcode** [comment]
@balazsorban44 I just updated my changes to reflect your comment. What do you think?

File: docs/docs/providers/reddit.md [hunk 2]
```diff
@@ -46,27 +46,19 @@ This Provider template only has a one hour access token to it and only has the "
...
-        email: null,
-      }
+    version: '2.0',
+    authorization: {
+      url: 'https://www.reddit.com/api/v1/authorize',
+      params: {
+        scope: 'identity',
+        duration: 'permanent',
+        response_type: 'code',
+      },
```
Comment (balazsorban44) [inline docs/docs/providers/reddit.md:59]:
```suggestion
    authorization: {
      params: {
        duration: 'permanent',
      },
```
Could be even simpler :grin:

**Review by balazsorban44 — COMMENTED** [review]
Since we deep-merge user configs, I could simplify this even more, but you were onto what I intended, thanks!

**Pull request merged**